libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/roadname/RoadNameProcessor.kt

```diff
@@ -16,6 +16,10 @@ internal object RoadNameProcessor {
         // Shield is being passed as null on purpose because this info is not yet available from
         // nav native. The ticket is being tracked here
         // https://github.com/mapbox/mapbox-navigation-native/issues/4325
-        return RoadNameResult.RoadNameLabel(road.name, null, road.shieldName)
+        val buffer = StringBuffer()
+        road.components.forEach { roadComponent ->
+            buffer.append(roadComponent.text)
+        }
+        return RoadNameResult.RoadNameLabel(buffer.toString(), null, null)
```

Comment:
```suggestion
        return RoadNameResult.RoadNameLabel(road.components.joinToString { it.text }, null, null)
```

Reply:
I forgot to mention that `joinToString` uses `", "` as a separator by default, which wasn't the case with your previous solution. But you can pass any separator you need to `joinToString`.

Reply:
@abhishek1508 what was the separator before?

Reply:
Earlier the current road name was one string like "Exit 12 I-5 South". But now they are broken into individual components like listOf("Exit", "12", "I-5", "South"). We need to concatenate all the strings. Looks like the separator should be " "